utils/shfun

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash -eu
+# This file is meant to be source'd by shell scripts
+
+SCRIPTDIR="$(dirname -- "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+. "${SCRIPTDIR}/shlog"
+
+[ -n "$(command -v python3)" ] && python="$(command -v python3)" || python="$(command -v python2)"
+export python
+
+trap interrupt_exception SIGINT
```

Comment:
cosmetic: should we move the definition of interrupt_exception before it is used?